BazelExtensions/xcode_configuration_provider.bzl

```diff
@@ -161,44 +173,134 @@ SourceOutputFileMapInfo = provider(
     },
 )
 
+def _objc_cmd_line(target, ctx, src, output_file_path):
+    """Returns flattened command line flags for ObjC sources
+    """
+    cc_toolchain = find_cpp_toolchain(ctx)
+    cc_feature_configuration = cc_common.configure_features(
+        ctx = ctx,
+        cc_toolchain = cc_toolchain,
+        requested_features = ctx.features,
+        unsupported_features = ctx.disabled_features,
+    )
+
+    copts = ctx.fragments.objc.copts + getattr(ctx.rule.attr, "copts", [])
+    non_arc_srcs = []
+
+    if hasattr(ctx.rule.attr, "non_arc_srcs"):
+        non_arc_srcs.extend([f for src in ctx.rule.attr.non_arc_srcs for f in src.files.to_list()])
+
+    cc_compile_variables = cc_common.create_compile_variables(
+        feature_configuration = cc_feature_configuration,
+        cc_toolchain = cc_toolchain,
+        user_compile_flags = copts,
+        source_file = src.path,
+        output_file = output_file_path,
+    )
+
+    base_args = cc_common.get_memory_inefficient_command_line(
+        feature_configuration = cc_feature_configuration,
+        action_name = ACTION_NAMES.objc_compile,
+        variables = cc_compile_variables,
+    )
+
+    cmd_line_args = []
+
+    # Arg pairs of the form '-flag value' to be skipped (breaks indexing if present)
+    arg_pairs_to_skip = ["-c"]
+    ignore_next = False
+
+    # Contains patterns in flag values that once detected should be skipped
+    #
+    # e.g.:
+    #
+    # DEBUG_PREFIX_MAP_PWD: gives an error "error: no such file or directory: 'DEBUG_PREFIX_MAP_PWD=.'" during indexing
+    #
+    arg_patterns_to_skip = ["DEBUG_PREFIX_MAP_PWD"]
```

Comment:
Unless there's something I'm missing these skips are necessary, hit compiler issues in Xcode when indexing was running without removing these flags